frontend/public/kubevirt/components/modals/create-vm-modal.js

```diff
@@ -15,27 +16,36 @@ export const openCreateVmWizard = ( activeNamespace, createTemplate = false ) =>
     namespaces: {
       resource: getResource(NamespaceModel),
     },
-    templates: {
-      resource: getResource(TemplateModel, {matchExpressions: [{key: TEMPLATE_TYPE_LABEL, operator: 'Exists' }]}),
+    userTemplates: {
+      resource: getResource(TemplateModel, {namespace: activeNamespace, prop: 'userTemplates', matchLabels: {[TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_VM}}),
+    },
+    commonTemplates: {
+      resource: getResource(VmTemplateModel, {namespace: 'openshift', prop: 'commonTemplates', matchLabels: {[TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_BASE}}),
     },
     networkConfigs: {
-      resource: getResource(NetworkAttachmentDefinitionModel),
+      resource: getResource(NetworkAttachmentDefinitionModel, {namespace: activeNamespace}),
     },
     storageClasses: {
       resource:  getResource(StorageClassModel),
     },
     persistentVolumeClaims: {
-      resource:  getResource(PersistentVolumeClaimModel),
+      resource:  getResource(PersistentVolumeClaimModel, {namespace: activeNamespace}),
     },
-  },(({namespaces}) => {
+  },(({namespaces, userTemplates, commonTemplates}) => {
       let selectedNamespace;
 
       if (namespaces && activeNamespace){
         selectedNamespace = namespaces.find(namespace => namespace.metadata.name === activeNamespace);
       }
 
+      let templates;
+      if (userTemplates && commonTemplates) {
```

Comment:
Can you please elaborate the use of these three `*Templates` props?
Why do we merge them here?
Why the `templates` is empty if just one of either user/common templates is missing?

Reply:
We need to split the templates queries to two because we need to load user-templates and common-templates. These templates are living in different namespaces. As a non-admin user we cannot create a query for all-namespaces, k8s will return forbidden instead of returning all available templates for user (k8s does not support ACL-filtering). Non-admin user has to query projects instead of namespaces but there is no possibility to create a query for all namespaces in all accessible projects.

I wanted to keep the change minimal and Create VM component from web-ui-components expects only one prop - templates so I will wait for both queries to finish and then pass them together to Create VM. When the prop is undefined it means that the query is still pending, when it is resolved to something (in our case array) it means that the query finished. 

We can change Create VM in web-ui-components to expect two props - user and common templates but I want to keep the change minimal for 1.4

Reply:
ok, thanks. Makes sense, especially the "waiting".